### ISSUE TITLE
 Layer Selection Fails on write.ml function

### ISSUE THREAD

**Issue by interdonatos** [magnanim/multinet#4]
Hi,

I was trying to select a single layer when writing down a 3 layer network, but the option seems not to be effective at all.
More specifically, trying to select just layer 3 with the command:

write.ml(net, "partialFile.mpx", format="multilayer", layers=3)

results in a network which appears to have the same edges as the input one (layers="3" gives the same result, while layers=character(3) returns a "cannot find layer").

Moreover, the resulting "LAYERS" section in the preamble has a weird syntax, e.g., from the one in the original net:

#LAYERS
1,UNDIRECTED
2,UNDIRECTED
3,UNDIRECTED

you obtain:

#LAYERS
3,3,UNDIRECTED
2,2,UNDIRECTED
1,1,UNDIRECTED
3,2,UNDIRECTED
3,1,UNDIRECTED
2,3,UNDIRECTED
2,1,UNDIRECTED
1,3,UNDIRECTED
1,2,UNDIRECTED

Thank you for your help,
Roberto

**Comment by magnanim** [comment]
Checked. There are two things:
1) the layer filtering is only supported for format graphml. I have added the support in the C++ code, it will appear in the next version of the library on CRAN.
2) the section is "correct" because the network is saved as a full multilayer. So, we must indicate two layers; the one where the edge starts and the one where the edge ends. But I can see it looks weird when the network is a multiplex. I have now added support for saving a multiplex format, also appearing in the next version.
